src/class.c

```diff
@@ -1205,6 +1205,17 @@ void ClassDeclaration::addLocalClass(ClassDeclarations *aclasses)
     aclasses->push(this);
 }
 
+/***************************************
+ * Return true if class is an InputRange.
+ * This is defined as having the members: empty, front, popFront.
+ */
+bool ClassDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
+            this->search(Loc(), Id::FpopFront) &&
+            this->search(Loc(), Id::Fempty));
+}
+
```

Comment:
Absolutely :) I can fix foreach if you want to. Do you want me defined predicates for other ranges aswell?